File: app/src/main/java/sk/brehy/PeopleAdapter.java
package sk.brehy;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;

import java.util.ArrayList;

public class PeopleAdapter extends ArrayAdapter<People> {

    PeopleAdapter(Context context, ArrayList<People> words) {
        super(context, 0, words);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        View listItemView = convertView;
        // View view = super.getView(position, convertView, parent);
        if (listItemView == null) {
            listItemView = LayoutInflater.from(getContext()).inflate(
                    R.layout.people_list, parent, false);
        }

        People current = getItem(position);

        TextView number = listItemView.findViewById(R.id.number);
        assert current != null;
        number.setText(current.getNumber());

        TextView name = listItemView.findViewById(R.id.name);
        name.setText(current.getName());

        return listItemView;
    }
}

File: app/src/main/java/sk/brehy/adapters/PeopleAdapter.kt
package sk.brehy.adapters

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import sk.brehy.R

data class People(var number: String, var name: String)

class PeopleAdapter(context: Context, words: List<People>)
    : ArrayAdapter<People>(context, 0, words as MutableList<People>) {
    @SuppressLint("ViewHolder")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val listItemView = LayoutInflater.from(context).inflate(R.layout.people_list, parent, false)
        val current = getItem(position)
        val number = listItemView.findViewById<TextView>(R.id.number)
        number.text = current!!.number
        val name = listItemView.findViewById<TextView>(R.id.name)
        name.text = current.name
        return listItemView
    }
}